File: app/src/main/java/com/example/mvp_sample/base/BaseApplication.java
package com.example.mvp_sample.base;

import android.app.Application;
import android.content.Context;

public class BaseApplication extends Application {

    private static Context mContext;

    @Override
    public void onCreate() {
        super.onCreate();
        mContext = getApplicationContext();
    }

    public static Context getContext() {
        return mContext;
    }
}

File: common/src/main/java/com/example/common/base/BaseApplication.kt
package com.example.common.base

import android.app.Application
import com.scwang.smart.refresh.footer.ClassicsFooter
import com.scwang.smart.refresh.header.ClassicsHeader
import com.scwang.smart.refresh.layout.SmartRefreshLayout

open class BaseApplication : Application() {
    companion object {
        private lateinit var instance: BaseApplication

        fun getInstance(): BaseApplication {
            return instance
        }
    }

    init {
        SmartRefreshLayout.setDefaultRefreshHeaderCreator { context, layout ->
            ClassicsHeader(context)
        }
        SmartRefreshLayout.setDefaultRefreshFooterCreator { context, layout ->
            ClassicsFooter(context)
        }
    }

    override fun onCreate() {
        super.onCreate()
        instance = this
    }
}